test/lib/heuristics/scheduling_test.rb

```diff
@@ -553,6 +553,20 @@ def test_sticky_in_scheduling
       refute_includes result[:routes].find{ |r| r[:activities].any?{ |stop| stop[:service_id] == 'service_6_1_1' } }[:vehicle_id], 'vehicle_0_'
     end
 
+    def test_skills_in_scheduling
+      vrp = VRP.lat_lon_scheduling_two_vehicles
+      result = OptimizerWrapper.wrapper_vrp('ortools', { services: { vrp: [:ortools] }}, TestHelper.create(vrp), nil)
+      assigned_route = result[:routes].find{ |r| r[:activities].any?{ |stop| stop[:service_id] == 'service_6_1_1' } }
+      assert_includes assigned_route[:vehicle_id], 'vehicle_0_' # default result
+
+      vrp = VRP.lat_lon_scheduling_two_vehicles
+      vrp[:vehicles][1][:skills] = [[:compatible]]
+      vrp[:services].find{ |s| s[:id] == 'service_6' }[:skills] = [:compatible]
+      result = OptimizerWrapper.wrapper_vrp('ortools', { services: { vrp: [:ortools] }}, TestHelper.create(vrp), nil)
+      assigned_route = result[:routes].find{ |r| r[:activities].any?{ |stop| stop[:service_id] == 'service_6_1_1' } }
+      refute_includes assigned_route[:vehicle_id], 'vehicle_0_'
```

Comment:
instead of refuting vehicle_0 (or at least additional to that), it would make more sense to check if the assigned route is the vehicle with the correct skill.

Reply:
if there is a chance that this service would get assigned to vehicle_1 even without the skills then this test would give false positives, right? 

(maybe this is already the case but I don't see it clearly, if that is the case a small note at the beginning of the test would be very useful) it might make sense to force the service to on a vehicle by "convenience" but then force it to be on the other "inconvenient" vehicle (with more cost) due to skills.

Reply:
That is exactly why we have "default result" (see your previous comment) : to ensure we are using this vehicle because of skills, and no other reason.